app/views/organisations/show.html.erb

```diff
@@ -7,11 +7,22 @@
 <div class="govuk-grid-row">
   <div class="govuk-grid-column-two-thirds-from-desktop">
     <%= govuk_summary_list do |summary_list| %>
-      <% @organisation.display_attributes.each do |attr, val| %>
-        <%= summary_list.row do |row|
-          row.key { attr.to_s.humanize }
-          row.value { simple_format(val.to_s, {}, wrapper_tag: "div") }
-        end %>
+      <% @organisation.display_attributes.each do |attr| %>
+
+        <% if current_user.data_coordinator? && attr[:editable] %>
+          <%= summary_list.row do |row|
+            row.key { attr[:name].to_s.humanize }
+            row.value { simple_format(attr[:value].to_s, {}, wrapper_tag: "div") }
+            row.action(visually_hidden_text: 'name', href: edit_organisation_path, html_attributes: { 'data-qa': "change-#{attr[:name]}" })
+          end %>
+        <% else %>
+          <%= summary_list.row do |row|
+            row.key { attr[:name].to_s.humanize }
+            row.value { simple_format(attr[:value].to_s, {}, wrapper_tag: "div") }
+            row.action()
+          end %>
+        <% end %>
```

Comment:
Should this logic not be in some kind of helper instead of the view?

Reply:
Happy to put it in one if you think I should. My reasoning for not was that doing so would pretty much move the entire view into a helper (since this block is pretty much all it has and it's more presentation than logic), so wouldn't really be isolating anything to unit test that isn't already being tested I don't think, not a strongly held opinion though!

Reply:
Yeah, I think this is fine, at least for now given the page and its current state.